bus-mapping/src/evm/opcodes/push.rs

```diff
@@ -0,0 +1,46 @@
+// Port this to a macro if possible to avoid defining all the PushN
+use super::Opcode;
+use crate::{
+    evm::GlobalCounter,
+    exec_trace::ExecutionStep,
+    operation::{container::OperationContainer, StackOp, RW},
+};
+use halo2::{arithmetic::FieldExt, plonk::ConstraintSystem};
+
+/// Number of ops that PUSH1 adds to the container & busmapping
+const PUSH1_OP_NUM: usize = 1;
+
+#[derive(Debug, Copy, Clone)]
+pub(crate) struct Push1;
+
+impl Opcode for Push1 {
+    fn gen_associated_ops(
+        &self,
+        exec_step: &mut ExecutionStep,
+        container: &mut OperationContainer,
+    ) -> usize {
+        let op = StackOp::new(
+            RW::WRITE,
+            GlobalCounter::from(exec_step.gc().0 + 1),
+            exec_step.stack_addr(),
+            // TODO: This could be more robust.
```

Comment:
Maybe implement some method like `peek` to get value from top of stack like [this](https://github.com/openethereum/openethereum/blob/main/crates/vm/evm/src/interpreter/stack.rs#L55-L57). But let's do this in future PRs.

Reply:
That's a good point. Will link this to the issue I opened about creating a `stack` struct.